mc/query/query.peg

```diff
@@ -0,0 +1,91 @@
+# Grammar for statement query language
+# just simple statements with no joins for now
+#
+# Compilation to query.peg.go:
+#  go get github.com/pointlander/peg
+#  peg -inline -switch query.peg
+
+package query
+
+type QueryParser Peg {
+    *ParseState
+}
+
+Grammar <- 'SELECT' WS Selector
+                    WS Source
+                    (WS Criteria)?
+                    (WS Limit)?
+                    WSX EOF
+
+Selector <- SimpleSelector   { p.setSimpleSelector() }
+          / CompoundSelector { p.setCompoundSelector() }
+          / FunctionSelector { p.setFunctionSelector() }
+
+SimpleSelector <- < SimpleSelectorOp > { p.push(text) }
+SimpleSelectorOp <- '*'
```

Comment:
I kind of want it to be strict, and also make it obvious what kind of operators we support.

Re: CompoundResult set containing an rset array instead of an interface (github doesn't let me reply there!!): The underlying QueryResult wouldn't work because of the hashing issues.